src/ai/ai.cpp

```diff
@@ -599,13 +599,13 @@ void ShipAI::flyTowards(glm::vec2 target, float keep_distance)
             keep_distance = 0.0;
 
         if (distance > keep_distance + owner->impulse_max_speed * 5.0f)
-            owner->impulse_request = 1.0f;
+            owner->setImpulseRequest(1.0f);
         else
-            owner->impulse_request = (distance - keep_distance) / owner->impulse_max_speed * 5.0f;
+            owner->setImpulseRequest((distance - keep_distance) / owner->impulse_max_speed * 5.0f);
```

Comment:
This retains the potential division by zero, but now appears to be clamped by `setImpulseRequest()`. The condition here might be changed to prevent division by zero, but it doesn't seem to cause other problems, so I kept the existing behavior.